### ISSUE TITLE
--nofetch: new CLI option to prevent remote downloads 

### ISSUE THREAD

**Issue by ry** [denoland/deno#1581]
This is the opposite of `--reload`... If you're in a deployed situation, you may have a $DENO_DIR prepared (using `--prefetch`) and you want to ensure you run exactly the code that is already cached and nothing else.

In this case `--nofetch` will prevent any downloads from happening during imports. The program should fail with non-zero error code if a given URL is not already cached in $DENO_DIR.

Maybe combine this with `--deny` (https://github.com/denoland/deno/issues/1580) since they will be used in similar situations.

**Comment by kitsonk** [comment]
With #1582 shouldn't it be `--noload` then?

**Comment by bartlomieju** [comment]
My 2 cents:
During development
`deno --load ...` - load all dependencies without running program. I hate that now if you run your program for first time and it throws error Deno will download deps again on next run.

Running production:
`deno --noload --noprompt ...` - don't download dependencies, don't display permission prompts, ie. error if anything is wrong. `--deny` is misleading IMHO since Deno requires explicit permission anyway.

**Comment by ry** [comment]
> deno --noload --noprompt ... - don't download dependencies, don't display permission prompts, ie. error if anything is wrong. --deny is misleading IMHO since Deno requires explicit permission anyway.

good point...

**Comment by kevinkassimo** [comment]
Or a flag `--prod`/`-P` as a short-hand to combine `--noload` and `--noprompt` (like what we did with `--allow-all` or `-A`)

**Comment by hayd** [comment]
RE #1582 it would be neat to do:
```
deno --prefetch deps.ts
deno --prefetch --nofetch main.ts  # okay, the naming is silly (you're really only compiling here)
```
to enforce that all your deps were defined in deps.ts.

**Comment by ry** [comment]
I want to err on the side of simplicity here - rather than enable every possible mode of usage. This is supposed to be fun and productive - understanding a bunch of nuanced flags is not fun.

Here's what I suggest:
```
deno --load https://deno.land/x/http/file_server.ts
```
This downloads (re-downloading if already cached) and compiles the source code - then it exits without running. This is the "hard reload" of the command-line.

Now to run this code, without accessing the internet. `--deny` 
```
deno --allow-net --deny https://deno.land/x/http/file_server.ts 
```
`--deny` both denies any permission prompts (which won't happen due to `--allow-net`) but also denies redownloading any imports. If a file is not cached it will exit with non-zero exit code.

**Comment by hayd** [comment]
so no --nofetch ?
It seems to me like a --compile and a --reload _are_ simpler. The original reason to use --reload vs the imo ambiguous --load is that it's standard nomenclature. ... as is --compile.

**Comment by bartlomieju** [comment]
Resolved in #2213 

CC @ry
